### ISSUE TITLE
Units of Measure Equality

### ISSUE THREAD

**Issue by thehoglet** [louthy/language-ext#419]
What should be the result here:

```
        var t1 = 123.Seconds();
        var t2 = 123.Seconds();

        object o1 = t1;
        object o2 = t2;

        var b3 = o1.Equals(o2);
```
?

I'm getting ```false```. Judging by this code from [Time.cs](https://github.com/louthy/language-ext/blob/master/LanguageExt.Core/DataTypes/UnitsOfMeasure/Time.cs#L35), the intent is to compare the .Value in each instance. If so, the runtime type check (Length?) is in error.

**Comment by bender2k14** [comment]
Definitely!

**Comment by thehoglet** [comment]
"Length" is present as an artefact in some of the other units implementations also. Beware!

Any idea if *units of measure* is going to be further developed? Is the system just an experiment? This morning, I implemented Force, Pressure, and Temperature for a project I'm working on. I included only the conversions I needed, though. It worked nicely and Units of Measure improved the API I'm working on. But, the experience informed me that implementing this for the entire universe of SI+conversions is a big job!

**Comment by louthy** [comment]
That’s a blatant copy and paste bug, it should be testing if the value is Time.  Fancy doing a pull request?

**Comment by bender2k14** [comment]
@louthy I already created PR #420 to fix this :)

**Comment by bender2k14** [comment]
> "Length" is present as an artefact in some of the other units implementations also. Beware!

Yep, I fixed the same issue in the other units of measure as well.

**Comment by louthy** [comment]
Fix deployed in v3.0.2 (thanks @bender2k14 )
